app/models/prisoner.rb

```diff
@@ -79,4 +79,11 @@ def self.search(query)
 
     where(noms_id: noms_ids, date_of_birth: dates_of_birth)
   end
+
+  def update_aliases(param_aliases)
+    aliases.delete_all
+    param_aliases.each do |alias_params|
+      aliases.create(alias_params)
+    end
```

Comment:
Maybe a couple minor things:

* no need to iterate over the ```param_aliases``` and create them all at once
```ruby
aliases.create(alias_params)
```
* possibly wrap the ```delete``` and ```insert``` inside a transaction